eZ/Publish/SPI/Search/FullTextFieldValue.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
+
+/**
+ * The wrapper for \eZ\Publish\SPI\Search\Field of the type \eZ\Publish\SPI\Search\FieldType\FullTextField.
+ */
+class FullTextFieldValue extends Field
+{
+    /**
+     * @param mixed $value
+     */
+    public function __construct($value)
+    {
+        parent::__construct('fulltext', (string) $value, new FieldType\FullTextField());
```

Comment:
@alongosz I know I implied that in
```
Ideally we should maybe:
- deprecate `getIndexData` and leave it as-is also returning fulltext data
- add `getFullTextData`, somewhat done
- add `<insert-name>` for non full text, _maybe `getFilterData,  `filter` is not the perfect domain word, but it’s the most applicable I can think of right now._

however to allow Solr/ES code to handle it somewhat uniform, return data for these two new methods should be somewhat in sync, or at least the wrapper `getFields()` should be able to handle the difference in format, so generic code in `mapFields()` can stay generic (e
```

However I also implied it could be mapped in Solr/ES.

Specific example: https://github.com/ezsystems/ezplatform-solr-search-engine/blob/master/lib/FieldMapper/ContentTranslationFieldMapper/BlockDocumentsContentFields.php#L78-Lundefined

instead of having `appendField()` below, the line linked to above can so something like ` $indexFields = $this->geSearchFields($field, $fieldDefinition);`, and the child Fulltext class extending this can override that protected method.


So you two can decide on whatever format is best to return from these two methods as long as `geSearchFields` is able to map it to a common format `mapFields()` can just use ;)